appcues/src/main/java/com/appcues/util/StringExt.kt

```diff
@@ -0,0 +1,11 @@
+package com.appcues.util
+
+import java.util.Locale
+
+fun String.toSlug() = lowercase(Locale.getDefault())
+    .replace("\n", " ")
+    .replace("[^a-z\\d\\s]".toRegex(), " ")
+    .split(" ")
+    .joinToString("-")
+    .replace("-+".toRegex(), "-")
+    .trimEnd('-')
```

Comment:
used when generating the user property name from a question label - matching existing web and ios logic

Reply:
whats the meaning of "[^a-z\\d\\s]"? on line 7? replacing special characters with white space?

Reply:
right, this is taken from https://stackoverflow.com/a/57298659 -- thats the idea though, normalize any survey question into a slugified name that is just normal chars or numbers separated by hyphens.